pkgs/stdenv/generic/check-meta.nix

```diff
@@ -312,17 +379,84 @@ let
     badPlatforms = platforms;
   };
 
-  checkMetaAttr = k: v:
-    if metaTypes?${k} then
-      if typeCheck metaTypes.${k} v then
+  # Type of a meta.problems.* value
+  problemTypes = with lib.types; {
```

Comment:
Module system types are not performant enough. This will have a significant impact on performance.
See https://github.com/NixOS/nixpkgs/pull/273935 for the work I did to bring performance to acceptable levels.

Reply:
I'm working on solving merge conflict son that I can rebase this PR, I saw this part with a comment stating it was better in terms of performance, I will keep the one from upstream.

Thanks for your review and for mentioning this.

Reply:
I believe I kept the change from the PR that you made, marking this as resolved

Reply:
What I mean is that module system types shouldn't be used at all in meta checks. They're too slow.
This is going to result in a performance regression.

Reply:
Sorry I think I minsunderstood your initial comment, I looked again at your PR and I should avoid using `lib.types` but use some of the types already declared in `meta-types.nix` (this would improve performance)

Reply:
@adisbladis  
I should add to `meta-types.nix` something like this, would this be alright : 
```nix
  77 + │   enum = values: assert isList values && all isString values; {
  78 + │     name = "enum<${concatStringsSep "," values}>";
  79 + │     verify = v: isString v && any (value: v == value) values;
  80 + │   };
```

And add a `let/in` statement for the `problemTypes` to remove the `lib.types`

Reply:
Yes that looks fine :)
`any (value: v == value) values` could be replaced by `elem v values`

Reply:
Thanks for the help, I think it is ready, I left the old way as a comment but will delete it.

Reply:
It seems `elem` is not defined, should I import it or define it ?

Reply:
I think I found it, it should be all right now.